Services/CustomsService.cs

```diff
@@ -0,0 +1,265 @@
+        };
+        return result2;
+    }
+
+    private CustomsResponse DoesCountryExist(string country1code, string country2code)
+    {
+        if (GetCountry(country1code) == default || GetCountry(country2code) == default)
+        {
+            var result = new CustomsResponse()
+            {
+                Message = "Country wasn't found in the list",
+                Success = false
+            };
+            return result;
+        }
+        else return null;
+    }
+
+    //Check if in one region
+    private bool CheckIfTheSameRegion(string c1, string p1, string c2, string p2)
+    {
+        if (c1 == c2)
+        {
+            var exceptionsForCountry = _customContext.PostalCodes.Where(c => c.Country.A2Code == c1);
+            var exceptionCodes = exceptionsForCountry.Select(b => b.Code).ToList();
+
+            foreach (var code in exceptionCodes)
+            {
+                if (p1.StartsWith(code) && p2.StartsWith(code))
+                {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    //Check does specific postal code belong to EUCU
+    private bool CheckIfEUCU(string country, string pCode, string city)
+    {
+        var countryCheck = CountryCheck(country);
+        if (!countryCheck)
+            return countryCheck;
+
+        var exceptionsForCountry = _customContext.PostalCodes.Where(c => c.Country.A2Code == country);
+
+        var zipcodeCheck = ZipcodeCheck(pCode, exceptionsForCountry);
+        if (!zipcodeCheck)
+            return zipcodeCheck;
+
+        var regionCheck = RegionCheck(pCode, exceptionsForCountry);
+        if (!regionCheck)
+            return regionCheck;
+
+        var cityCheck = CityCheck(pCode, city, exceptionsForCountry);
+        if (!cityCheck)
+            return cityCheck;
+
+        return true;
+    }
+
+    private bool CityCheck(string pCode, string city, IQueryable<PostalCode> exceptionsForCountry)
+    {
+        foreach (var exceptionForCountry in exceptionsForCountry)
+        {
+            if (exceptionForCountry.Type == PostalCodeType.City
+            && exceptionForCountry.Code == pCode
+            && exceptionForCountry.City == city)
+            {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    private bool RegionCheck(string pCode, IQueryable<PostalCode> exceptionsForCountry)
+    {
+        foreach (var exceptionForCountry in exceptionsForCountry)
+        {
+            if (exceptionForCountry.Type == PostalCodeType.Region && pCode.StartsWith(exceptionForCountry.Code))
+                return false;
+        }
+        return true;
+    }
+
+    private bool ZipcodeCheck(string pCode, IQueryable<PostalCode> exceptionsForCountry)
+    {
+        // postleitzahl check
+        // if (exceptionCodes.Contains(pCode) )
+        //    return false; // customs ist true
+        foreach (var exceptionForCountry in exceptionsForCountry)
```

Comment:
I think this function is an optimal case for LINQ, but this is more of an FYI and not something you have to change 😄

Reply:
var result = !exceptionsForCountry.Any(exceptionForCountry => exceptionForCountry.Code == pCode && exceptionForCountry.Type == SpecialCaseType.Zipcode);